src/transport/message_handler.rs

```diff
@@ -0,0 +1,210 @@
+use crate::events::{Event, EventHandler};
+use crate::transport::msgs::{LSPSMessage, Prefix, RawLSPSMessage, LSPS_MESSAGE_TYPE};
+use bitcoin::secp256k1::PublicKey;
+use lightning::ln::peer_handler::CustomMessageHandler;
+use lightning::ln::wire::CustomMessageReader;
+use lightning::log_info;
+use lightning::util::logger::Logger;
+use std::collections::HashMap;
+use std::convert::{TryFrom, TryInto};
+use std::io;
+use std::ops::Deref;
+use std::sync::{Arc, Mutex};
+
+/// A trait used to implement a specific LSPS protocol
+/// The messages the protocol uses need to be able to be mapped
+/// from and into LSPSMessages.
```

Comment:
nit: let's tick and link all non-local objects here and elsewhere (i.e., everything that you're currently not describing should be linked)
```suggestion
/// from and into [`LSPSMessages`].
```